src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1485,6 +1485,38 @@ ss::future<> do_fetch(op_context& octx) {
 }
 } // namespace
 
+namespace testing {
+ss::future<> do_fetch(op_context& octx, model::fetch_read_strategy strategy) {
+    switch (strategy) {
+    case model::fetch_read_strategy::polling: {
+        // first fetch, do not wait
+        co_await fetch_topic_partitions(octx).then([&octx] {
+            return ss::do_until(
+              [&octx] { return octx.should_stop_fetch(); },
+              [&octx] { return fetch_topic_partitions(octx); });
+        });
+    } break;
```

Comment:
I have never seen this before. `break` outside of the curly braces?

Reply:
Normally I add the curly braces to give each case-statement its own local scope. So from;
```cpp
case foo:
// body
break;
```
to
```cpp
case foo: {
// body
} break;
```
Happy to switch to a different pattern though if its not a great way of going about it.

Reply:
I always put the break inside the curly braces. I didn't even know that was valid syntax.